basis/io/directories/directories.factor

```diff
@@ -93,6 +85,18 @@ M: object copy-file
 : copy-files-into ( files to -- )
     '[ _ copy-file-into ] each ;
 
+! Moving and renaming files
+HOOK: move-file io-backend ( from to -- )
+
+M: object move-file ( from to -- )
+    dupd copy-file delete-file ;
```

Comment:
@mrjbq7: "Is this implementation here so that your new backend works?"

No, I just thought it would be nice to have a generic implementation.